GVFS/GVFS.Common/GVFSEnlistmentStatistics.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            if (this.trackedFilesCount == 0)
+            {
+                return 0;
+            }
+
+            return (double)this.modifiedPathsCount / this.trackedFilesCount;
+        }
+
+        /// <summary>
+        /// Take a string representing a file path on system and pull out the upper-most directory from it as a string, or '/' if it is in the root
+        /// </summary>
+        /// <param name="path">The path to a file to parse for the top level directory containing it</param>
+        /// <returns>A string containing the top level directory from the provided path, or '/' if the path is for an item in the root</returns>
+        private string ParseTopDirectory(string path)
+        {
+            int whackLocation = path.IndexOf('/');
+            if (whackLocation == -1)
+            {
+                return "/";
+            }
+
+            return path.Substring(0, whackLocation);
+        }
+
+        /// <summary>
+        /// Categorizes a list of paths given as strings by mapping them to the top level directory in their path
+        /// Modifies the directoryTracking dictionary to have an accurate count of the files underneath a top level directory
+        /// </summary>
+        /// <remarks>
+        /// The distinction between files and directories is important --
+        /// If the path to a file doesn't contain a '/', then that means it falls within the root
+        /// However if a directory's path doesn't contain a '/', it counts towards itself for hydration
+        /// </remarks>
+        /// <param name="paths">An enumerable containing paths as strings</param>
+        /// <param name="directoryTracking">A dictionary used to track the number of files per top level directory</param>
+        private void CategorizePaths(IEnumerable<string> paths, Dictionary<string, int> directoryTracking, ref int count, string parentDirectory, bool isFile)
+        {
+            foreach (string path in paths)
+            {
+                // Ensure all paths have the same slashes
+                string formattedPath = path.Replace('\\', '/');
+                if (formattedPath.StartsWith("/"))
+                {
+                    formattedPath = formattedPath.Substring(1);
+                }
+
+                if (formattedPath.StartsWith(parentDirectory))
+                {
+                    count++;
+
+                    if (isFile)
+                    {
+                        string topDir = this.ParseTopDirectory(this.TrimDirectoryTarget(formattedPath, parentDirectory));
+                        directoryTracking[topDir] = directoryTracking.ContainsKey(topDir) ? directoryTracking[topDir] + 1 : 1;
+                    }
+                    else
+                    {
+                        if (!parentDirectory.Equals(formattedPath))
+                        {
+                            string topDir = this.TrimDirectoryTarget(formattedPath, parentDirectory);

```

Comment:
maybe indicate via the name that this is a relative path (relative to the `parentDirectory`)

Reply:
So all paths are relative to the src directory of the repository. This method's purpose is to chomp off the target directory from the front of the path since we already know it begins with that (see line 135)